File: java/2292 벌집.java
import java.util.*;

public class Main {
	
	public static void main(String[] args) {
		Scanner scn = new Scanner(System.in);
		
		int n = scn.nextInt();
		int layer = 1;
		int total = 1;
		
		while (n > total)
		{
			total += layer * 6;
			layer++;
		}
		
		System.out.println(layer);
		
		
		scn.close();

	}

}

File: C++/2292 벌집.cpp
#include <iostream>
using namespace std;

int main()
{
	int n;
	cin >> n;
	int layer = 1; // 층 수
	int total = 1; // 층의 최대 주소

	while (n > total)
	{
		total += layer * 6;
		layer++;
	}

	cout << layer;

	
	return 0;
}
